docs/source/ESPEasy/FunctionBlocks.rst

```diff
@@ -1,2 +1,166 @@
 ESP Easy Function Blocks
 ************************
+
+
+The workflow and flow of data in ESPEasy can be seen as a number of functional blocks:
+
+* **Plugin** - A piece of code designed to interact with hardware, like sensors. See: :ref:`Plugin List`
+* **Task** - An instance of a plugin. It may yield some data collected from the connected hardware and send it to a connected controller or trigger an event to be processed in rules.
+* **Controller** - Receives data from a task to send to some service. (e.g. MQTT broker, flash storage, etc.)
+* **Rules** - Scripting language to act on events in ESPEasy.
+* **Notifications** - Extra options to notify a user, like a buzzer or sending an email.
+* **Scheduler** - The internal heart beat of ESPEasy.
+
+  
+
+Plugin
+======
+
+A plugin is a piece of code, typically designed to interact with some hardware.
+There are some exceptions like the "Dummy" plugin, which is only meant to allow a user to store data to be processed.
+
+A plugin may also implement a few commands specific for that plugin.
+
+For example a command to set some configuration on the connected hardware or interact with some display.
+
+Not all plugins are included in each build, due to limits of flash size on lots of ESP devices.
+
+See: :ref:`Plugin List`
+
+
+Task
+====
+
+A Task is an instance of a Plugin.
```

Comment:
This doesn't explain why there is a tab Devices, that is showing all... Tasks. Some explanation on that could be useful.

Reply:
It was intended to be a short introduction to what blocks / concepts there are in ESPEasy.

I think I failed miserably on the "short" part already, so some extra explaining here can also be added I guess...

Reply:
It does explain a major part of the concept, so that's at least an improvement.